### PR TITLE
Answer:21

### PR DESCRIPTION
**Pull request by Sanjar1304**
Following your advice I managed and learned how to make PR for per task. Thanks so much Thomas !!!

### CODE REVIEW

File: apps/angular/anchor-scrolling/src/app/foo.component.ts [hunk 1]
```diff
@@ -1,15 +1,44 @@
...
+      section 2
+      <button class="m-2 block w-fit rounded-md border border-red-500 p-4">
+        Section 1
+      </button>
+    </div>
   `,
 })
-export class FooComponent {}
+export class FooComponent {
+  scroller = inject(ViewportScroller);
```
Comment (svenson95) [inline apps/angular/anchor-scrolling/src/app/foo.component.ts:36]: `scroller` is unsed?

File: apps/angular/anchor-scrolling/src/app/home.component.ts [hunk 2]
```diff
@@ -1,20 +1,34 @@
...
-      <nav-button href="#top">Scroll Top</nav-button>
+      <nav-button>Scroll Top</nav-button>
     </div>
   `,
 })
-export class HomeComponent {}
+export class HomeComponent {
+  scroller = inject(ViewportScroller);
+
+  scrollTo(anchor: string) {
```
Comment (svenson95) [inline apps/angular/anchor-scrolling/src/app/home.component.ts:27]: You have the same code for `scrollTo` in `FooComponent`, you should refactor this in a service for example.

**Comment by svenson95** [comment]
Good job 👍 but i'd recommend you to checkout the authors solutions as well, there is a way to add this functionality without writing any code in the class.

File: apps/angular/anchor-scrolling/src/app/foo.component.ts (hunk 1 above)
Comment (Sanjar1304) [reply apps/angular/anchor-scrolling/src/app/foo.component.ts]: removed unused scroller

File: apps/angular/anchor-scrolling/src/app/home.component.ts (hunk 2 above)
Comment (Sanjar1304) [reply apps/angular/anchor-scrolling/src/app/home.component.ts:24]: here also removed unused scroller

Comment (Sanjar1304) [reply apps/angular/anchor-scrolling/src/app/home.component.ts]: Add service and scrolling function moved to service file

**Pull request closed without merging**